frontend/src/app/modules/user/profile-edit-form/profile-edit-form.component.ts

```diff
@@ -0,0 +1,92 @@
+import {
+  Component, EventEmitter, OnInit, Output, Input
+} from '@angular/core';
+import { FormControl, FormGroup, Validators } from '@angular/forms';
+import { Router } from '@angular/router';
+import { User } from '../../../models/user/user';
+import { countries } from '../data/countries';
+import { Country } from '../models/country';
+
+declare global {
+  interface JQuery {
+    modal(action: string): void;
+  }
+}
+
+@Component({
+  selector: 'app-profile-edit-form',
+  templateUrl: './profile-edit-form.component.html',
+  styleUrls: ['./profile-edit-form.component.sass']
+})
+export class ProfileEditFormComponent implements OnInit {
+  form!: FormGroup;
+  genders: { key: boolean; text: string; }[] = [{ key: false, text: 'Male' }, { key: true, text: 'Female' }];
+  countries: Country[] = countries;
+  standartIcon: string = '../../../../assets/images/standartIcon.png';
+
+  @Input()
+  user: User;
+
+  @Output()
+  updatedUser = new EventEmitter<User>();
+
+  constructor(private _router: Router) { }
+
+  ngOnInit() {
+    this.form = new FormGroup({
+      userName: new FormControl(this.user.userName, [
+        Validators.required,
+        Validators.pattern('[A-Za-z0-9.-_]{6,20}')
+      ]),
+      email: new FormControl(this.user.email, [
+        Validators.required,
+        Validators.email
+      ]),
+      description: new FormControl(this.user.description, [
+        Validators.required,
+        Validators.maxLength(150)
+      ]),
+      birthday: new FormControl(this.user.birthday),
+      iconURL: new FormControl(this.user.iconURL),
+      gender: new FormControl(this.user.gender),
+      country: new FormControl(this.user.country)
+    });
+  }
+
+  get email() {
+    return this.form.get('email')!;
+  }
+
+  get userName() {
+    return this.form.get('userName')!;
+  }
+
+  get description() {
+    return this.form.get('description')!;
+  }
```

Comment:
There is a way to get rid of these getters and setters, use ngModel. Read about it here.
In this code, we have only 3 fields(getter), but imagine if we have 20 fields so we should have 20 getters.

https://angular.io/guide/forms